infra/k8s/README.md

```diff
@@ -1,39 +1,97 @@
 
-## Background
 
-This document is explaining how to execute a Testground test plan locally, while Testground targets a Kubernetes cluster on Digital Ocean for execution of all tasks. Images are built locally by Testground via Docker, and pushed to Docker Hub. Redis is to be installed manually on Kubernetes, prior to running any test plans.
+## Set up infrastructure with kops
 
-## Introduction
+1. [Configure your AWS credentials](https://docs.aws.amazon.com/cli/)
 
-1. `kubectl` - `kubectl` is a command line interface for running commands against Kubernetes clusters. It sends commands to the Kubernetes master, and all `kubectl` commands are run on the developer's machine in this tutorial. `kubectl` has to be configured with access parameters so that it has access to any given Kubernetes cluster.
+2. Create a bucket for kops state. This is similar to Terraform state bucket.
+
+```
+aws s3api create-bucket \
+    --bucket kops-backend-bucket \
+    --region eu-central-1 --create-bucket-configuration LocationConstraint=eu-central-1
+```
 
-2. `helm` - `helm` is a package manager for Kubernetes. If you have configured `kubectl` to a given cluster, then `helm` also has access to it, and you can run `helm` commands from your local machine towards it.
+3. Pick up a cluster name, and set zone and kops state store
 
-## Setup Kubernetes Cluster on Digital Ocean
+```
+export NAME=my-first-cluster-kops.k8s.local
+export KOPS_STATE_STORE=s3://kops-backend-bucket
+export ZONES=eu-central-1a
+```
 
-1. Create a Kubernetes cluster on Digital Ocean, with the necessary number of workers.
+4. Create The cluster
 
-Note: There is a limit of 100 pods per worker, so calculate the number of workers you need accordingly.
+```
+kops create cluster \
+  --zones $ZONES \
+  --master-zones $ZONES \
+  --master-size m4.xlarge \
+  --node-size m4.xlarge \
+  --node-count 2 \
+  --networking cni \
+  --name $NAME \
+  --yes
+```
 
-2. Configure your `kubectl` context and authentication with Digital Ocean. An easy way to do it is to download the config provided by Digital Ocean and placing it at `~/.kube/config`. Confirm that `kubectl get nodes` shows the number of nodes in your cluster.
+5. Wait for `master` node to appear in `kubectl get nodes` - it will be in `NotReady` state as we haven't installed any Networking CNI yet.
+
+6. Install Flannel
+
+```
+kubectl apply -f ./flannel.yml
+```
+
+7. Wait for all nodes to appear in `kubectl get nodes` with `Ready` state.
+
+8. Install CNI-Genie
+
+```
+kubectl apply -f ./genie-plugin.yaml
+```
+
+9. Install Weave
 
 ```
-$  kubectl get nodes
-NAME                  STATUS   ROLES    AGE   VERSION
-pool-jyub1rb95-snl2   Ready    <none>   60m   v1.16.2
-pool-jyub1rb95-snl5   Ready    <none>   60m   v1.16.2
-pool-jyub1rb95-snl6   Ready    <none>   60m   v1.16.2
+kubectl apply -f ./weave.yml
 ```
 
+10. Destroy the cluster when you're done working on it
+
+```
+kops delete cluster $NAME --yes
+```
```

Comment:
Can we create a script for all this?

Reply:
I.e., a "cluster-create.sh" and a "cluster-delete.sh"?

Reply:
Yes, we should automate it further. The reason I haven't done so far is, because we need to wait for given states on Kubernetes before running commands (for example wait for master to come up as `NotReady` before installing `flannel`, wait for `flannel` to be installed, before installing `genie`, etc.)

Still this needs to be simpler.